### DISCUSSION TITLE
rocm-gdb in the official repos?

### DISCUSSION THREAD

**Discussion by simark** [rocm-arch/rocm-arch#984] [Ideas]
Hi!

Is there a reason why rocm-gdb is not in the official repos?

The support for debugging in the AMD KFD module wasn't in the upstream Linux kernel until [somewhat recently](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/log/drivers/gpu/drm/amd/amdkfd/kfd_debug.c), so I can imagine that it would have been useless to have an official rocm-gdb package before.

FWIW, I tested changing the PKGBUILD to point to the 6.0.2 release, and it builds and runs fine (although only lightly tested).

I'm a gdb developer and also contribute to ROCgdb, so I'd be happy to help get rocm-gdb in the official repos, if that's what you want.

Thanks,

Simon

**Comment by tpkessler** [comment]
Hi Simon!

That's a very nice surprise. Thank you for pointing me to the kernel commit. Indeed. the main reason why rocm-gdb was not in the official repos was the lack of support by the mainline linux kernel. Do you have suggestions for `configure` flags before I add to the Arch repos?

Cheers,
Torsten

**Comment by simark** [comment]
What you have already seems to work.  But since you ask:

* If it works, I would remove `--disable-nls`, to let people have GDB in whatever language they want (not that I use it, and I don't even know the state of translations)
* `$CHOST` in `--enable-targets="$CHOST,amdgcn-amd-amdhsa"` is unnecessary, I think.  Since you don't specify a `--target`, the default target will be the same as the host (i.e. `x86_64-pc-linux-gnu`), and is automatically included.  If you pass `--enable-targets="amdgcn-amd-amdhsa"`, the support for that target will be in addition to the default target.
* There's an approximately 0% chance that `--with-babeltrace` is useful to anybody, I would remove it (it's been on my todo list for a while to remove it from upstream GDB).
* I see that you don't use `--disable-binutils`, which means binutils binaries are built and shipped.  I think that's fine, since there is a chance that the ROCm release of binutils-gdb contains some changes to better support inspecting AMD GPU binaries, that vanilla binutils tools don't have.  You could think of packaging it separately as a `rocm-binutils` package, but it's not a big deal.
* You could add `--disable-gprof`.
* You used `--disable-gdbserver`, which I think is fine.  There's no support currently for debugging ROCm programs with ROCgdb through gdbserver, so it would be pointless to ship it.
* `--disable-shared` seems unnecessary to me.

I then looked at the built package itself, there is a bunch of stuff (headers, libraries) that come from bfd, binutils, and other projects in the binutils-gdb source tree (I know, it's a bit of a mess with all those projects in the same tree).  I think it would make sense to run "make install" in the gdb directory only.  This is what the gdb package does:

https://gitlab.archlinux.org/archlinux/packaging/packages/gdb/-/blob/main/PKGBUILD?ref_type=heads#L53

**Reply by tpkessler** [reply]
Thank you very much for your suggestions! I've updated the configure flags and moved `rocm-gdb` to the official repos.

**Reply by simark** [reply]
Thanks!
